packages/open-api-gateway/src/project/open-api-gateway-ts-project.ts

```diff
@@ -142,7 +182,7 @@ export class OpenApiGatewayTsProject extends TypeScriptProject {
       // client won't be visible for the first install in this project's post synthesize step, so we use a local
       // workspace as a workaround.
       if (this.package.packageManager === NodePackageManager.PNPM) {
-        new YamlFile(this, "pnpm-workspace.yaml", {
+        this.pnpmWorkspace = new YamlFile(this, "pnpm-workspace.yaml", {
```

Comment:
This will still error if the user has already defined a `pnpm-workspace.yaml` file. I see two options:

a.) Allow the user to pass in a YamlFile representing the pnpmWorkspace. We check to make sure it is being written to `pnpm-workspace.yaml` and then append the `typescriptCodeGenDir` as a package.
b.) Add documentation to the `README.md` stating that this package will generate and export a PNPMWorkspace `YamlFile` and this file should be used to make any changes to the workspace. If a user decides to byo, synthesis will fail.

Reply:
Cool I feel like (a) is a bit nicer - will update :)

Reply:
Chatted offline and going for (b) :)